dna-src/zomes/my_zome/code/src/lib.rs

```diff
@@ -164,20 +164,27 @@ mod my_zome {
     }
 
     #[zome_fn("hc_public")]
-    fn create_goal(goal: Goal) -> ZomeApiResult<GetResponse<Goal>> {
+    fn create_goal(goal: Goal, parent_address: Address) -> ZomeApiResult<GetResponse<Goal>> {
         let app_entry = Entry::App("goal".into(), goal.clone().into());
-        let _ = hdk::commit_entry(&app_entry)?;
+        let entry_address = hdk::commit_entry(&app_entry)?;
 
-        // link each new goal to the anchor
+        // link new goal to the anchor
         let anchor_address = Entry::App(
             "anchor".into(), // app entry type
             "goals".into() // app entry value
         ).address();
 
         hdk::link_entries(&anchor_address, &app_entry.address(),  "anchor->goal", "")?;
 
+        // if a parent address was provided, link the goal with its parent
+        // if parent_address != None {
+        if true {
```

Comment:
Syntax you can use here is 

match parent_address {
  Some(address) => inner_edge_fn(address),
  None => ()
}

Reply:
Aha @Connoropolous thanks! I thought I could do something with `match` and `Some()` but couldn't get there by myself. I'll play with these suggestions 😄